### ISSUE TITLE
exp 08: fine-tune YOLOv8s on CARLA dataset + cross-map validity probe

### ISSUE THREAD

**Issue by joeljose** [joeljose/SkyCop#19]
## Why

Exp 06 established: pretrained COCO YOLOv8s is 4.6% recall on our aerial frames. Exp 07 produced 5000 in-domain training frames. Exp 08 closes the detection-milestone loop by fine-tuning and measuring the delta.

Critical addition from grill Q1: same-map eval has a silent-overfit blind spot. A detector that memorises Town10HD_Opt's textures/lighting would produce a stellar mAP@0.5 on the exp 05 holdout (also Town10HD_Opt) while failing on any other map. Exp 08 adds a small cross-map probe to distinguish real generalisation from memorisation.

### Dataset preparation

- Build Ultralytics YOLO dataset YAML from \`output/dataset/carla_pursuit/dataset_manifest.json\`:
  - \`train\`: image paths from \`train_runs\` (3500 frames)
  - \`val\`: image paths from \`val_runs\` (1500 frames)
  - \`nc: 1\`, \`names: ['vehicle']\`
- Ultralytics expects \`labels/\` alongside \`images/\` in YOLO convention; verify or symlink.
- Exp 05 holdout (Town10HD_Opt) stays frozen — never in train/val during training.

### Cross-map validity probe (new in exp 08)

- Capture ~100 frames on **Town03** using \`skycop.cv.capture.run_capture\` with a fresh seed (seed=900, distinct from any exp 07 or exp 05 seed).
- Single pursuit, ClearNoon. Same adaptive-altitude scene otherwise.
- Output: \`output/eval/carla_eval_town03/{images, labels, manifest.json}\`.
- NEVER used during training.

### Training

- \`skycop/cv/training.py\` — thin wrapper around \`ultralytics.YOLO.train\`, signature keeps options injectable for tests + config.
- Start from \`yolov8s.pt\` (auto-downloads) — same pretrained weights as exp 06 baseline.
- Hyperparameters (locked):
  - \`epochs=30\`, \`patience=10\` (early stopping on val mAP@0.5)
  - \`imgsz=640\`, \`batch=-1\` (Ultralytics auto-sizer; falls back to 8 if OOM)
  - \`optimizer='AdamW'\`, cosine LR (Ultralytics default), warmup 3 epochs
  - \`mosaic=0\` (aerial frames don't benefit from 4-image splicing)
  - \`degrees=0\` (aerial is pre-oriented)
  - Keep \`hsv_h/s/v\` and \`scale/translate\` (genuine augmentations for aerial)
  - \`seed=42\`, \`deterministic=True\`
  - \`half=True\` (fp16)

### Evaluation (two numbers, different stories)

1. **Same-map mAP** — evaluate fine-tuned weights on exp 05 holdout (200 frames, Town10HD_Opt). Headline CV-03 number.
2. **Cross-map mAP** — evaluate fine-tuned weights on Town03 mini-holdout (~100 frames, Town03). Generalisation probe.

Reported as \`output/eval/fine_tuned_metrics.json\`:
\`\`\`json
{
  \"model\": \"output/weights/best.pt\",
  \"fine_tuned_same_map\":  { /* same shape as baseline_metrics */ },
  \"fine_tuned_cross_map\": { /* same shape, different eval dir */ },
  \"pretrained_baseline_same_map\": 0.050,  // from exp 06 for side-by-side
  \"interpretation_gap_threshold\": 30,
  \"notes\": [\"single-map training, see progress.md for memorisation caveat\"]
}
\`\`\`

Interpretation rules (documented in script + progress.md):
- **same_map < 30%** — something is broken (data pipeline, label alignment, wrong classes). Debug before reporting.
- **same_map ≥ 70% AND cross_map < 30%** — memorisation. Caveat loudly; plan multi-map training for a follow-up.
- **same_map ≥ 70% AND cross_map ≥ 65%** — genuine generalisation. Celebrate cautiously.

### Deliverables

- \`scripts/08_finetune_yolo.py\`
- \`skycop/cv/training.py\`
- \`configs/training.yaml\`
- \`output/weights/best.pt\` (fine-tuned) + \`output/weights/run_v1/\` (ultralytics run artefacts)
- \`output/eval/carla_eval_town03/\` (cross-map holdout, 100 frames, regenerable)
- \`output/eval/fine_tuned_metrics.json\`

### Docs

- \`docs/progress.md\` — exp 08 row + numbers + cross-map gap
- \`docs/design.md\` D-11 — \"measure generalisation via cross-map probe\" as the durable practice

## Acceptance criteria

- [ ] Training completes within 3 hours on RTX 4050 fp16 without OOM
- [ ] \`output/weights/best.pt\` exists; \`output/eval/fine_tuned_metrics.json\` has both same-map and cross-map entries
- [ ] Same-map mAP@0.5 ≥ 30% (minimum bar — anything lower means pipeline broken)
- [ ] Cross-map mAP@0.5 ≥ 30% OR explicit caveat in progress.md stating memorisation-suspected
- [ ] \`make test\` + \`make lint\` green
- [ ] Unit tests for dataset-YAML construction + metrics-JSON shape

## Out of scope

- VisDrone warm-start — defer; revisit if cross-map number is catastrophic.
- Multi-map training data generation — exp 07-v2 if the memorisation verdict requires it.
- In-loop inference with fine-tuned weights — that's exp 09.
- Tracking / fingerprinting / mission loop — later experiments.

## Risks

- **Ultralytics API changes.** Pinning \`ultralytics\` in pyproject.toml would be wise (currently unpinned). Follow-up chore if it breaks.
- **CUDA non-determinism** may produce slightly different mAP numbers between identical seed runs; documented in metrics JSON.
- **Training OOM on RTX 4050 + CARLA server competing for VRAM.** If training and CARLA conflict, bring CARLA down before starting training via \`docker compose stop carla-server\`.
